### PR TITLE
FIX Job arg in CLI

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-queuedjobs/issues/495

### CODE REVIEW

File: src/Tasks/ProcessJobQueueTask.php [hunk 1]
```diff
@@ -8,7 +8,6 @@
 use SilverStripe\PolyExecution\PolyOutput;
 use SilverStripe\PolyExecution\PolyOutputLogHandler;
 use Symbiote\QueuedJobs\Services\AbstractQueuedJob;
-use Symbiote\QueuedJobs\Services\QueuedJob;
```
Comment (emteknetnz) [inline src/Tasks/ProcessJobQueueTask.php:11]: Unused import

File: src/Tasks/ProcessJobQueueTask.php [hunk 2]
```diff
@@ -64,11 +63,14 @@ protected function execute(InputInterface $input, PolyOutput $output): int
...
         $job = $input->getOption('job');
-        if ($job && strpos($job, '-')) {
-            // Run from a single job
-            $parts = explode('-', $job ?? '');
-            $id = $parts[1];
-            $service->runJob($id);
+        if ($job) {
+            // Accepts an ID (e.g. "123"), optionally prefixed with anything up to a dash
+            // which covers a job cache file name (e.g. "queuedjob-123") and a leading dash (e.g. "-123")
+            if (!preg_match('#^(?:[^-]*-)?(\d+)$#', $job, $matches)) {
```
Comment (emteknetnz) [inline src/Tasks/ProcessJobQueueTask.php:69]: The old check was `if ($job && strpos($job, '-'))` so it only passed when a dash sat past position 0, and the id then came out of explode('-', $job)[1]. --job=734803 has no dash in it so it fell straight through to a full queue run, and --job=-734803 gives strpos() === 0 which is falsy as well.

File: tests/ProcessJobQueueTaskTest.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        parent::setUp();
+        Injector::inst()->registerService(new RecordingQueuedJobService(), QueuedJobService::class);
+    }
+
+    public static function provideJobOption(): array
+    {
+        return [
+            'numeric ID' => ['734803', 734803],
+            'job cache file name' => ['queuedjob-734803', 734803],
+            'ID with a leading dash' => ['-734803', 734803],
```
Comment (ChloeSartorelli) [inline tests/ProcessJobQueueTaskTest.php]: Please add a scenario with multiple `-`, e.g. `'queued-job-prefix-734803'`

**Review by ChloeSartorelli — CHANGES_REQUESTED** [review]
Looks good, just one test scenario to add for edge-case purposes.

File: tests/ProcessJobQueueTaskTest.php (hunk 3 above)
Comment (emteknetnz) [reply tests/ProcessJobQueueTaskTest.php]: Added that - exposed a issue with regex so I've fixed `ProcessJobQueueTask::execute()` too

Also refactored the dataProvider so it labels the args and splits over multple lines

**Review by ChloeSartorelli — APPROVED** [review]
LGTM
